dotnet/src/InternalUtilities/src/Data/VectorStoreRecordPropertyReader.cs

```diff
@@ -337,6 +338,61 @@ public string GetJsonPropertyName(string dataModelPropertyName)
         return this._jsonPropertyNamesMap.Value[dataModelPropertyName];
     }
 
+    /// <summary>
+    /// Get the <see cref="VectorStoreRecordVectorProperty"/> for the given property name.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">When there is no match.</exception>
+    public VectorStoreRecordVectorProperty GetVectorProperty<TRecord>(VectorSearchOptions<TRecord>? searchOptions)
+    {
+        if (searchOptions is not null)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            string? propertyName = searchOptions.VectorPropertyName;
```

Comment:
Unless I'm mistaken, the VectorPropertyName already refers to the **storage** name (in the database) rather than to the **model** name (name of the CLR property) - we should confirm (this is important for when the names differ. If so, then we should not be looking it up in VectorProperties as below, but just using that directly.

If so, then this method can simply accept the expression lambda directly, rather than VectorSearchOptions. At that point I'd also consider making it generic for looking up any property, not just vector properties - we'd be able to use that e.g. for the hybrid search additional property.

Reply:
It depends on the connector. I want it to accept `VectorSearchOptions` as I want to have the logic for dealing with obsolete property in one place (less code, easier to delete in the future)

Reply:
> It depends on the connector [...]

This kind of thing really, really shouldn't depend on the connector - either VectorPropertyName should refer to the model name or to the storage name, but whatever it is should be consistent across connectors...

Anyway, since we're removing this it's not so important.